app/cdap/components/GridTable/index.tsx

```diff
@@ -206,9 +199,17 @@ export default function GridTable({ handleTransformationUpload, storeData }) {
         setShowBreadCrumb={setShowBreadCrumb}
         showBreadCrumb={showBreadCrumb}
         columnType={dataprep.types[selectedColumn]}
-        submitMenuOption={(option, datatype) =>
-          !transformationOptions.includes(option) ? onMenuOptionSelection(option, datatype) : null
-        }
+        hasSimilarType={checkAllColumnsHasSimilarType(Object.values(dataprep.types))}
```

Comment:
try to merge allColumnsType and hasSimilarType into single prop

Reply:
removed hasSimilarType prop and used allColumnsType as similarTypeCheck and getting type as well